### PR TITLE
Adapt data ingestion to the new memory interface

### PR DESCRIPTION
**Pull request by k-boikov**
### Background
Data ingestion was not working for a long time. Tests will come with https://github.com/Significant-Gravitas/Auto-GPT/pull/3602

### Changes
Adapt the data ingestion to the new memory interface.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run `black .` and `isort .` against my code to ensure it passes our linter.

### CODE REVIEW

File: data_ingestion.py [hunk 1]
```diff
@@ -21,22 +24,30 @@ def configure_logging():
...
+        files = list_files(directory, config)
         for file in files:
-            ingest_file(file, memory, args.max_length, args.overlap)
+            ingest_file(os.path.join(config.workspace_path, file), memory)
     except Exception as e:
         logger.error(f"Error while ingesting directory '{directory}': {str(e)}")
 
 
+def setup_workspace(cfg):
+    workspace_directory = Path(__file__).parent / "autogpt" / "auto_gpt_workspace"
```
Comment (Pwuts) [inline data_ingestion.py]: I don't know when this became the _de facto_ workspace path, but it doesn't seem right to me.

Also duplicates hard-coded default workspace path elsewhere, which is impractical. Maybe create a constant in the workspace module with the default path?

File: data_ingestion.py [hunk 2]
```diff
@@ -69,19 +80,22 @@ def main() -> None:
     )
     args = parser.parse_args()
 
+    # Setup workspace
```
Comment (Pwuts) [inline data_ingestion.py]: This is unnecessary considering the name of the function

File: data_ingestion.py (hunk 1 above)
Comment (k-boikov) [reply data_ingestion.py]: Agree, reworked.

File: autogpt/workspace/workspace.py [hunk 3]
```diff
@@ -48,6 +48,13 @@ def make_workspace(cls, workspace_directory: str | Path, *args, **kwargs) -> Pat
             The path to the workspace directory.
 
         """
+        # TODO: have this directory live outside the repository (e.g. in a user's
+        #   home directory) and have it come in as a command line argument or part of
+        #   the env file.
+        if workspace_directory is None:
+            workspace_directory = Path(__file__).parent.parent / "auto_gpt_workspace"
```
Comment (Pwuts) [inline autogpt/workspace/workspace.py:55]: Still resolves to `autogpt/auto_gpt_workspace`. I would prefer moving it to `data/workspace` or `data/auto_gpt_workspace`, or else restoring it to its original location `auto_gpt_workspace` (not in `autogpt/`).

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Pull request closed without merging**